src/charm.py

```diff
@@ -316,8 +326,16 @@ def _handle_auth_config(self, _):
         """Event handler for forward_auth config changes."""
         self._sync_all_resources()
 
+    def _on_peers_changed(self, _):
+        """Event handler for whenever peer topology changes."""
+        self._sync_all_resources()
+
     def _on_remove(self, _):
         """Event handler for remove."""
+        # if there are still units left, skip removal
```

Comment:
Suggestion: this feels novel enough that we should describe it in more detail

```suggestion
        """Event handler for remove.

        The objective of this handler is to remove all application-scoped resources when the application is being scaled
        to 0 or removed.  We intentionally do not put this removal action behind a leader guard (eg, behind 
        `if self.unit.is_leader()`) for the reasons discussed 
        [here](https://github.com/canonical/istio-ingress-k8s-operator/issues/16).
        """
        # skip if we are not scaling to 0/removing the application
```